### PR TITLE
[Workflows] Document all test-run data options in the YAML editor

### LINKED ISSUE #7350: [Security][Workflows][9.5 & Serverless] Filter historical workflow executions by time range when replaying or testing a step
## Summary

When testing a workflow step or replaying a workflow run against a past execution, the execution-selection combobox now has an adjacent time-range picker (EuiSuperDatePicker, defaulting to last week through now). Opening replay from a specific execution anchors the range so that run&#x27;s start time is the end of the range. If the user changes the range so the currently selected execution falls out of the list, the selection is cleared and the form is kept in a consistent not-ready state instead of showing a stale or stuck editor. The server-side execution and step-execution list APIs gained optional startedAfter/startedBefore query parameters to support this.

**Why this needs docs:** New UI control (time-range filter) and matching API parameters for an existing historical-replay feature; the diff (EuiSuperDatePicker, startedAfter/startedBefore params, useWorkflowExecutions changes) matches the PR&#x27;s stated user-facing behavior.

## Resources

- PR [#266706](https://github.com/elastic/kibana/pull/266706) — [One Workflow] Add date range filter to the historical execution selection

<details><summary>Screenshots from PR</summary>

screenshot
</details>

## Availability

| Channel | Details |
|---------|---------|
| **Stack** | v9.5.0 |
| **Serverless** | May 25–May 29 |
| **Feature flag** | None — active by default |

---
*Created with [Docs Quest Scanner](https://github.com/florent-leborgne/docs-quest-scanner) by @nastasha-solomon*

## Suggested edits

[Monitor workflow execution](https://www.elastic.co/docs/explore-analyze/workflows/authoring-techniques/monitor-workflows) > View execution history
- **What the docs say:** To review past runs, select the Executions tab, then click each run to see detailed logs. Workflow runs can be Pending, In progress, Completed, or Failed.
- **What to add:** The page doesn&#x27;t describe the historical execution picker used when testing a step or replaying a workflow run, so the new time-range filter next to that combobox (defaulting to last week through now, and anchored to the selected run when opened from an execution) isn&#x27;t documented anywhere a user would look for execution-history help. Add a short section explaining how to narrow the historical execution list by time range and how the anchor behavior works when replaying from a specific run. Applies from 9.5.0 and in serverless (week of May 25-29).

### PR DESCRIPTION
**Pull request by nastasha-solomon**
## Summary

Restructures the **Supply test input** section on the [Use the YAML editor](/explore-analyze/workflows/authoring-techniques/use-yaml-editor.md) page — renamed **Provide data for a test run** — so all three ways to supply test data are actually explained, not just one:

- Previously, only the **Historical** option (reuse data from a previous execution) was documented; **Document** and **Manual** were listed but never explained.
- Adds a dedicated subsection for each option: **Use a document from Elasticsearch**, **Type your own JSON**, and **Reuse data from a previous run**.
- Documents the time-range filter next to the historical execution picker (defaults to the last week).
- Replaces "input data" with plain language ("data") throughout the section.
- Aligns the test-run dialog name (**Test workflow**) across the Use the YAML editor, Monitor workflow execution, and Event-driven triggers pages — it was previously called the **Run workflow** dialog in one place.
- Updates cross-references on the Monitor workflow execution and Event-driven triggers pages to point to the renamed section.

Fixes https://github.com/elastic/docs-content/issues/7350.

## Previews

* [Use the YAML editor | Provide data for a test run](https://docs-v3-preview.elastic.dev/elastic/docs-content/pull/7646/explore-analyze/workflows/authoring-techniques/use-yaml-editor#workflows-supply-test-input): New section explaining all three test-run data options (Document, Manual, Historical).
* [Monitor workflow execution | View execution history](https://docs-v3-preview.elastic.dev/elastic/docs-content/pull/7646/explore-analyze/workflows/authoring-techniques/monitor-workflows#workflows-execution-history): Documents all **Executions** tab statuses (**Pending**, **Queued**, **Running**, **Waiting**, **Success**, **Error**, **Canceled**, **Timed out**, **Skipped**) with definitions that match the current UI labels. **Queued** is tagged `stack: ga 9.5+`.

## Generative AI disclosure

1. Did you use a generative AI (GenAI) tool to assist in creating this contribution?
- [x] Yes - Claude Code
- [ ] No

### CODE REVIEW

File: explore-analyze/workflows/authoring-techniques/use-yaml-editor.md [hunk 1]
```diff
@@ -53,9 +53,31 @@ Every workflow execution is either a test run or a production run. Understanding
...
+
+Select **Document**, then search for and pick a document from an {{es}} index. The workflow runs against that document's actual field values, which is useful for confirming your steps behave correctly with real data before you rely on a live trigger.
+
+### Type your own JSON [workflows-supply-test-input-manual]
+
+Select **Manual**, then enter JSON directly in the dialog. This is useful for testing a specific scenario, such as an edge case or a field value that's hard to find in your existing data.
+
+Select **Historical**, then, under **Select execution**, search or pick a previous run from the list. A time-range filter next to the picker defaults to the last week, so you can narrow a long execution history down to the runs you care about. After you select an execution, click **Run** to start a test run using that execution's data.
+
+Reusing data this way is useful when you want to reproduce an issue or validate a workflow change against real data from a prior run, without tracking down the original event or re-entering values by hand.
```
Comment (tinnytintin10) [inline explore-analyze/workflows/authoring-techniques/use-yaml-editor.md:87]: @nastasha-solomon should there be a heading here for selecting inputs from previous runs? something like "### Reuse data from a previous run"

**Review by tinnytintin10 — CHANGES_REQUESTED** [review]
@nastasha-solomon left one inline comment on a structural issue. beyond that, one accuracy gap: the tabs in the dialog depend on the workflow's triggers, so all three options aren't always shown. 

document only appears when the workflow has a manual trigger, and alert triggered workflows get an alert tab that this section doesn't mention (only event gets a callout). as written, "choose one of the following" implies all three are always there. a one line fix covers it: note that the options depend on the workflow's triggers, and mention the alert tab alongside the event one. cc @yngrdyn 

thanks for this 🙏

File: explore-analyze/workflows/authoring-techniques/use-yaml-editor.md (hunk 1 above)
Comment (nastasha-solomon) [reply explore-analyze/workflows/authoring-techniques/use-yaml-editor.md:87]: Yeah, I can add that - thanks!

Comment (nastasha-solomon) [reply explore-analyze/workflows/authoring-techniques/use-yaml-editor.md:87]: Header added and [additional comments](https://github.com/elastic/docs-content/pull/7646#pullrequestreview-4850847303) addressed in [`6168d4b` (this PR)](https://github.com/elastic/docs-content/pull/7646/commits/6168d4be92e1057ea6800fc19c17323fefa75651),

File: explore-analyze/workflows/authoring-techniques/use-yaml-editor.md [hunk 2]
```diff
@@ -53,9 +53,37 @@ Every workflow execution is either a test run or a production run. Understanding
...
+
+### Use a document from {{es}} [workflows-supply-test-input-document]
+
+Select **Document**, then search for and pick a document from an {{es}} index. The workflow runs against that document's actual field values, which is useful for confirming your steps behave correctly with real data before you rely on a live trigger.
+
+### Type your own JSON [workflows-supply-test-input-manual]
+
+Select **Manual**, then enter JSON directly in the dialog. This is useful for testing a specific scenario, such as an edge case or a field value that's hard to find in your existing data.
+
+### Reuse data from a previous run [workflows-supply-test-input-historical]
```
Comment (natasha-moore-elastic) [inline explore-analyze/workflows/authoring-techniques/use-yaml-editor.md:79]: Should this have a section-level applies tag?

File: explore-analyze/workflows/authoring-techniques/use-yaml-editor.md [hunk 3]
```diff
@@ -53,9 +53,37 @@ Every workflow execution is either a test run or a production run. Understanding
...
\ No newline at end of file
+You can use the `execution.isTestRun` context variable in your workflow YAML to change behavior during testing. For example, you can choose to skip sending a real notification during a test run.
+
+## Provide data for a test run [workflows-supply-test-input]
+
+When you click **Run** {icon}`play` in the editor, the **Test workflow** dialog asks what data to run the workflow against. Which options appear depends on the workflow's triggers. Common options include:
+
+- **Document**: Use a real document from {{es}}. Appears when the workflow has a [manual trigger](/explore-analyze/workflows/triggers/manual-triggers.md).
+- **Manual**: Type your own JSON.
+- **Historical**: Reuse the data from a previous run. {applies_to}`stack: ga 9.5+` {applies_to}`serverless: ga`
```
Comment (natasha-moore-elastic) [inline explore-analyze/workflows/authoring-techniques/use-yaml-editor.md:69]: Nit: I think we usually place the tag at the start of the bullet?

File: explore-analyze/workflows/authoring-techniques/use-yaml-editor.md [hunk 4]
```diff
@@ -53,9 +53,37 @@ Every workflow execution is either a test run or a production run. Understanding
...
+
+If you open the dialog from a specific past execution, the time range is anchored to that run. The end of the range is the execution's start time, and the start of the range is one week earlier. That keeps the selected run in the list. If you change the time range so the selected execution falls outside it, the selection clears and **Run** stays unavailable until you pick another execution.
+
+After you select an execution, click **Run** to start a test run using that execution's data.
+
+Reusing data this way is useful when you want to reproduce an issue or validate a workflow change against real data from a prior run, without tracking down the original event or re-entering values by hand.
+
+For [event-driven](/explore-analyze/workflows/triggers/event-driven-triggers.md) workflows, the dialog also includes an **Event** tab where you can pick a real event to test against. Refer to [Test a workflow with a real event](/explore-analyze/workflows/triggers/event-driven-triggers.md#event-driven-triggers-test). For [alert-triggered](/explore-analyze/workflows/triggers/alert-triggers.md) workflows, the dialog includes an **Alert** tab where you can pick a real alert to test against.
```
Comment (natasha-moore-elastic) [inline explore-analyze/workflows/authoring-techniques/use-yaml-editor.md:89]: I think this paragraph (line 89) should go at the end of the  **Provide data for a test run**  section rather than inside **Reuse data from a previous run** because it talks about other ways to choose data to test against, and these tabs (Event  and  Alert) sit at the same level as Document, Manual, and Historical.

File: explore-analyze/workflows/authoring-techniques/use-yaml-editor.md [hunk 5]
```diff
@@ -53,9 +53,37 @@ Every workflow execution is either a test run or a production run. Understanding
...
+
+### Type your own JSON [workflows-supply-test-input-manual]
+
+Select **Manual**, then enter JSON directly in the dialog. This is useful for testing a specific scenario, such as an edge case or a field value that's hard to find in your existing data.
+
+### Reuse data from a previous run [workflows-supply-test-input-historical]
+
+Select **Historical**, then, under **Select execution**, search or pick a previous run from the list. A time-range filter next to the picker defaults to the last week, so you can narrow a long execution history down to the runs you care about.
+
+If you open the dialog from a specific past execution, the time range is anchored to that run. The end of the range is the execution's start time, and the start of the range is one week earlier. That keeps the selected run in the list. If you change the time range so the selected execution falls outside it, the selection clears and **Run** stays unavailable until you pick another execution.
```
Comment (natasha-moore-elastic) [inline explore-analyze/workflows/authoring-techniques/use-yaml-editor.md:83]: > If you open the dialog from a specific past execution

Can we describe how to do this? I'm looking at a previous execution for a workflow in a test env, and can't find a way to open the **Test workflow** dialog from it

**Review by natasha-moore-elastic — APPROVED** [review]
A couple of minor suggestions, otherwise LGTM 🚀

File: explore-analyze/workflows/authoring-techniques/use-yaml-editor.md (hunk 3 above)
Comment (nastasha-solomon) [reply explore-analyze/workflows/authoring-techniques/use-yaml-editor.md:69]: I've seen both, but have recently preferred putting it at the end because it's less disruptive that way.

File: explore-analyze/workflows/authoring-techniques/use-yaml-editor.md (hunk 4 above)
Comment (nastasha-solomon) [reply explore-analyze/workflows/authoring-techniques/use-yaml-editor.md]: Makes sense - moved in https://github.com/elastic/docs-content/pull/7646/commits/4149b4e76a76fc85a13a49d41b693fba74ba63e0..

File: explore-analyze/workflows/authoring-techniques/use-yaml-editor.md (hunk 2 above)
Comment (nastasha-solomon) [reply explore-analyze/workflows/authoring-techniques/use-yaml-editor.md:83]: Addressed in https://github.com/elastic/docs-content/pull/7646/commits/4149b4e76a76fc85a13a49d41b693fba74ba63e0.

File: explore-analyze/workflows/authoring-techniques/use-yaml-editor.md (hunk 5 above)
Comment (nastasha-solomon) [reply explore-analyze/workflows/authoring-techniques/use-yaml-editor.md]: Addressed in https://github.com/elastic/docs-content/pull/7646/commits/4149b4e76a76fc85a13a49d41b693fba74ba63e0.

File: explore-analyze/workflows/authoring-techniques/monitor-workflows.md [hunk 6]
```diff
@@ -43,7 +43,9 @@ Every execution begins with a trigger entry that shows how the run started, such
 
 ## View execution history [workflows-execution-history]
 
-To review past runs, select the **Executions** tab, then click each run to see detailed logs. Workflow runs can be `Pending`, `In progress`, `Completed`, or `Failed`. 
+To review past runs, select the **Executions** tab, then click each run to see detailed logs. Workflow runs can be `Pending`, `In progress`, `Completed`, or `Failed`.
```
Comment (talboren) [inline explore-analyze/workflows/authoring-techniques/monitor-workflows.md]: waiting too. i think it's redundant to list the states here as they might changed in the future?

**Review by talboren — APPROVED** [review]
lgtm. can we add screenshots maybe?

File: explore-analyze/workflows/authoring-techniques/monitor-workflows.md (hunk 6 above)
Comment (nastasha-solomon) [reply explore-analyze/workflows/authoring-techniques/monitor-workflows.md]: Was `Waiting` added in a recent release, or was this status just missed in docs when the **Executions** tab was GA'd in 9.4?

Comment (talboren) [reply explore-analyze/workflows/authoring-techniques/monitor-workflows.md]: I think we did have the waitForInput primitive in 9.4 already, so I think it was just missed

Comment (nastasha-solomon) [reply explore-analyze/workflows/authoring-techniques/monitor-workflows.md]: Added the `Waiting` status and status explanations in https://github.com/elastic/docs-content/pull/7646/changes/9829514586ca1d75d1df49012aba4ab03338774d. Claude also suggested changing the `In progress` status to `Running`, which is the current label showing in `main`.

Comment (nastasha-solomon) [reply explore-analyze/workflows/authoring-techniques/monitor-workflows.md]: A few additional findings about docs for statuses that display in the **Execution** tab:

- In 9.2, https://github.com/elastic/kibana/pull/236683 modified labels for several status badges:
  - `Completed` → `Success`
  - `Failed` → `Error`
  - `Waiting for input` → `Waiting`
- Several UI labels were missing docs (fixed in https://github.com/elastic/docs-content/pull/7646/changes/801b0bc128f670becf955bbd0447bdee08e9a9d9):
  - `Queued` was added in 9.5 via https://github.com/elastic/kibana/pull/268596.
  - `Canceled` - Label first added via https://github.com/elastic/kibana/pull/228259 in 9.2 as `Cancelled`. https://github.com/elastic/kibana/pull/236683 renamed the badge from `Cancelled` to `Canceled`. The engine value stayed `cancelled`.
  - `Timed out`- Added in 9.2 via https://github.com/elastic/kibana/pull/236308.
  - `Skipped` - Added in 9.2 via https://github.com/elastic/kibana/pull/228259.

**Review by natasha-moore-elastic — APPROVED** [review]

**Review by tinnytintin10 — APPROVED** [review]
LGTM 🚀  thanks @nastasha-solomon 👑
